src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -329,9 +339,51 @@ protected void getChanges2(ChangeEventSourceContext context,
                     for (Pair<String, String> entry : tabletPairList) {
                         final String tabletId = entry.getValue();
                         curTabletId = entry.getValue();
-                        YBPartition part = new YBPartition(tabletId);
+                        YBPartition part = new YBPartition(entry.getKey() /* tableId */, tabletId, false /* colocated */);
+
+                      OpId cp = offsetContext.lsn(part);
+
+                      if (taskContext.shouldEnableExplicitCheckpointing()
```

Comment:
Change to accomodate explicit checkpointing + tablet splitting --> for the scenarios where we haven't received a callback for split tablets yet.